transform/mattermost-analytics/models/marts/product/nps/fct_nps_score.sql

```diff
@@ -0,0 +1,94 @@
+with user_metrics as (
+    select activity_date,
+    server_id AS server_id,
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='promoters',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_promoters_daily'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='detractors',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_detractors_daily'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='passives',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_passives_daily'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='nps_users',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_nps_users_daily'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='promoters_last90d',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_promoters_last90d'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='detractors_last90d',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_detractors_last90d'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='passives_last90d',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_passives_last90d'
+  ) }},
+    {{ dbt_utils.pivot(
+      'user_role',
+      dbt_utils.get_column_values(ref('int_user_nps_score_spined'), 'user_role'),
+      agg='sum',
+      then_value='nps_users_last90d',
+      quote_identifiers=False,
+      prefix='count_',
+      suffix='_nps_users_last90d'
+  ) }}
+    FROM
+    {{ ref('int_user_nps_score_spined') }}
+    group by activity_date
+    , server_id
+)
+SELECT a.*,
+    {{ dbt_utils.generate_surrogate_key(['a.activity_date', 'a.server_id']) }} AS daily_server_id,
+    b.server_version AS server_version,
+    a.count_user_promoters_daily + a.count_team_admin_promoters_daily + a.count_system_admin_promoters_daily AS count_promoters_daily,
```

Comment:
It would be nice to have a few tests for sanity checks. I.e. daily detractors + passives + promoters = count of NPS users.

Reply:
Added tests to the model